### ISSUE TITLE
Clearing inline content makes editor momentarily collapse

### ISSUE THREAD

**Issue by seaneking** [SimplaElements/simpla-text#54]
Since ProseMirror denounces state updates, the placeholder for simpla-text doesn't show up until a little while after content is cleared. This is fine for block elements, but for inline simpla-texts the element's height collapses to 0, causing a jarring page reflow.

Related (but more minor) issue: entering content in an empty element 'pushes' the placeholder along for a bit before it's cleared.

Both issues would be solved if we could prevent the placeholder state updates from denouncing.

**Comment by bedeoverend** [comment]
>denounce, _verb_
     - publicly declare to be wrong or evil.

Erm...do you by chance mean debouncing? 😉 

There is [inDOMChange](https://github.com/ProseMirror/prosemirror-view/blob/master/src/domchange.js#L60) but it's undocumented so I'd be hesitant to use it.

The other, probably more viable one, is to find a way for the placeholder to predict the next state change based on incoming events, that way it will update with the DOM, rather than asynchronously due to a change in state.

I'll dig into this some more

**Comment by seaneking** [comment]
It's not private though? Unless ProseMirror doesn't use the underscore convention

**Comment by bedeoverend** [comment]
Don't know whether they do or not, which means I'd err on the side of undocumented -> can break in a patch. Easiest way to solve this is probably just bring it up in their discussion forums / raise an issue on GitHub.

**Comment by seaneking** [comment]
Yep, I'd say so. File an issue on whether inDom thing is public interface or not

**Comment by bedeoverend** [comment]
Actually, that's not gonna work 🤔 The placeholder plugin listens to changes _after_ the state has changed on the view - at that point checking a few things including if the state is empty before saying it should update the state with a placeholder decoration.

I'd say the way to do it is to try add the decoration before the updated state has been applied to the view. Not sure how to do that though.

**Comment by seaneking** [comment]
Hmm okay. It does feel pretty buggy, but only _breaking_ with inline text elements that rely on their content for height. And even then, only breaking minorly in edit mode. So I wouldn't dive too deep on it now, but worth tinkering with a fix for.
